python/paddle/fluid/layers/nn.py

```diff
@@ -2523,6 +2523,12 @@ def get_decay_momentum(momentum_init, decay_steps, decay_rate):
     check_type_and_dtype(input, 'input', Variable,
                          ['float16', 'float32', 'float64'], 'batch_norm')
     dtype = helper.input_dtype()
+
+    has_reserve_space = False
+    if data_layout == 'NHWC' and os.environ.get(
```

Comment:
`os.environ.get('FLAGS_cudnn_batchnorm_spatial_persistent')` would return a string or bool. So if `FLAGS_cudnn_batchnorm_spatial_persistent=false`, this condition would return true before string `"false"` is true in Python.

Reply:
Done.Thanks.